common/types/mapper/thrift/shared.go

```diff
@@ -4234,6 +4234,7 @@ func FromRequestCancelWorkflowExecutionRequest(t *types.RequestCancelWorkflowExe
 		WorkflowExecution: FromWorkflowExecution(t.WorkflowExecution),
 		Identity:          &t.Identity,
 		RequestId:         &t.RequestID,
+		Cause:             &t.Cause,
```

Comment:
I don't know if it'll matter or not, and we're doing this for other fields too, but:
should this actually be "if empty, nil, else `&t.Cause`"?

thrift differentiates between `nil` and `""` but proto doesn't, so I wonder if this means we're passing back a bunch of empty-but-not-nil fields.  might explain why some of our logs / CLI outputs / etc are showing zero valued data...

---

but anyway, not blocking at all, just an observation / side question, and maybe something for someone to tackle in a later PR of some kind.

Reply:
Cause is string field not pointer so it will still have emtpy value and can't be nil, it is good question in general, but I think leaving cause as empty string is okay either way since it is just metadata